eth/downloader/downloader.go

```diff
@@ -917,6 +913,23 @@ func (d *Downloader) findAncestor(p *peerConnection, remoteHeader *types.Header)
 		localHeight = d.lightchain.CurrentHeader().Number.Uint64()
 	}
 
+	// Check the validity of peer from which the chain is to be downloaded
+	if d.ChainValidator != nil {
+		_, err := d.IsValidPeer(d.getFetchHeadersByNumber(p))
+		if errors.Is(err, whitelist.ErrMismatch) {
+			return 0, err
+		}
+
+		// Don't validate the peer against whitelisted milestones until the different of
+		// our local height and remote peer's height is less than `maxValidationThreshold`
+		if errors.Is(err, whitelist.ErrNoRemote) && localHeight >= remoteHeight-d.maxValidationThreshold {
```

Comment:
Maybe we can move `localHeight >= remoteHeight-d.maxValidationThreshold` condition inside the if statement?

Reply:
Done, thanks!